application/src/test/java/uk/nhs/hee/tis/revalidation/controller/DoctorsForDBControllerTest.java

```diff
@@ -272,6 +280,38 @@ void shouldReturnDoctorsByGmcId() throws Exception {
         .andExpect(status().isOk());
   }
 
+  @Test
+  void shouldReturnAutocompleteResultsForProgrammeName() throws Exception {
+    final var fieldNameParam = "programmeName";
+    final var inputParam = "general prac";
+    final var dbcsParam = List.of("1-AIIDWI");
+    var result =
+        List.of(
+            "General Practice Salford and Trafford",
+            "General Practice York"
+        );
+
+    var returnedValue = when(recommendationElasticSearchService.getAutocompleteResults(
+        fieldNameParam,
+        inputParam,
+        dbcsParam
+    )).thenReturn(result);
+
+    this.mockMvc.perform(get(GET_AUTOCOMPLETE)
+            .param(AUTOCOMPLETE_FIELD, fieldNameParam)
+            .param(INPUT, inputParam)
+            .param(DESIGNATED_BODY_CODES, new String[]{"1-AIIDWI"}))
+        .andExpect(status().isOk())
+        .andExpect(content().json(mapper.writeValueAsString(result)));
+  }
+
+  @Test
+  void shouldRejectAutocompleteResultsForIvalidField() throws Exception {
```

Comment:
Misleading name suggests that the problem is the `fieldName` param... the absence of input and/or dbcs causes the HTTP 400.

`@RequestParam(required = true, ...)` is the default

Reply:
Ah this was from an old implementation that didn't work anyway, thanks for catching!